types/constants.ts

```diff
@@ -0,0 +1,69 @@
+import { ChainNames, NativeToken, Token } from "./handler";
+
+export declare const chainIDList: Record<string, string>;
+export declare const extraRpcs: Record<ChainId, string[]>;
+
+export type ChainId<T extends string | number = number> = T extends keyof typeof chainIDList ? (typeof chainIDList)[T] : T;
+
+export const permit2Address = "0x000000000022D473030F116dDEE9F6B43aC78BA3";
+export const nftAddress = "0xAa1bfC0e51969415d64d6dE74f27CDa0587e645b";
+export const LOCAL_HOST = "http://127.0.0.1:8545";
+
+export const networkIds: Record<ChainNames[keyof ChainNames], ChainId> = Object.fromEntries(
+  Object.entries(chainIDList).map(([id, name]) => {
+    const chainId = parseInt(id);
+    const chain = name.charAt(0).toUpperCase() + name.slice(1);
+    return [chain, chainId];
+  })
+);
+
+export const networkNames: ChainNames = Object.fromEntries(
+  Object.entries(networkIds).map(([name, id]) => {
+    const chainName = name.charAt(0).toUpperCase() + name.slice(1);
+    return [id, chainName];
+  })
+);
+
+export const networkRpcs: Record<ChainId, string[]> = Object.fromEntries(
+  Object.entries(networkIds).map(([, value]) => {
+    const chainRpcs = extraRpcs[value] || [];
+    return [value, chainRpcs];
+  })
+);
+
+export const tokens: Record<ChainId, Record<string, Token>> = {
```

Comment:
Hardcoding is a bad idea as we plan to support any token. 

Consider my implementation on pay.UBQ.fi

It's super simple it just saves the address as the key name in the localStorage with its associated values (like decimals and symbol)

Reply:
`tokens` is not actually used anywhere in this codebase pav, I'm exporting them from here because they are used in some UIs and I thought that since this will likely be added to near all repos it would be a good thing to export which can be extended in the importing codebase if need be.